tests/components/compensation/test_sensor.py

```diff
@@ -0,0 +1,234 @@
+"""The tests for the integration sensor platform."""
+
+from homeassistant.components.compensation.const import CONF_PRECISION, DOMAIN
+from homeassistant.components.compensation.sensor import ATTR_COEFFICIENTS
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import (
+    ATTR_UNIT_OF_MEASUREMENT,
+    EVENT_HOMEASSISTANT_START,
+    EVENT_STATE_CHANGED,
+    STATE_UNKNOWN,
+)
+from homeassistant.setup import async_setup_component
+
+
+async def test_linear_state(hass):
+    """Test compensation sensor state."""
+    config = {
+        "compensation": {
+            "test": {
+                "name": "compensation",
+                "entity_id": "sensor.uncompensated",
+                "data_points": [
+                    [1.0, 2.0],
+                    [2.0, 3.0],
+                ],
+                "precision": 2,
+                "unit_of_measurement": "a",
+            }
+        }
+    }
+
+    assert await async_setup_component(hass, DOMAIN, config)
+    assert await async_setup_component(hass, SENSOR_DOMAIN, config)
+    await hass.async_block_till_done()
+
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    entity_id = config[DOMAIN]["test"]["entity_id"]
+    hass.states.async_set(entity_id, 4, {})
+    await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.compensation")
+    assert state is not None
+
+    assert round(float(state.state), config[DOMAIN]["test"][CONF_PRECISION]) == 5.0
+
+    assert (
+        state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+        == config[DOMAIN]["test"][ATTR_UNIT_OF_MEASUREMENT]
```

Comment:
Maybe explicitly compare against `"a"` instead to make it easier to read what's expected.

Reply:
good suggestion.  Makes sense.